### PR TITLE
Update to 3.5.3

### PR DESCRIPTION
**Pull request by Arishamays1**
# `Matplotlib` version bump to 3.5.3

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -38,6 +38,7 @@ outputs:
         - setuptools
         - setuptools_scm >=4
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:39]: Should be `setuptools_scm >=7`, see https://github.com/matplotlib/matplotlib/blob/318b2348f7127efc3e28aadebf762aae32b6343c/setup.py#L306

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
   name: matplotlib-suite
   version: {{ version }}
 
 source:
   url: https://github.com/matplotlib/matplotlib/archive/v{{ version }}.tar.gz
-  sha256: c81325d1b0d822ef75fd71d243f98478cdca1f800a838a2a1bc8e4e17b8df93c
+  sha256: 9d6ee7190fde3cb0c6285d462a0258c9735a72c7b532321a3983e05076aa41dc
 
 build:
   number: 0
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:15]: Now the minimal supported python version is **3.8,** see https://github.com/matplotlib/matplotlib/blob/318b2348f7127efc3e28aadebf762aae32b6343c/setup.py#L11
So we have to skip py<38 for matplotlib-base and matplotlib

File: recipe/meta.yaml [hunk 3]
```diff
@@ -38,6 +38,7 @@ outputs:
         - setuptools
         - setuptools_scm >=4
         - setuptools_scm_git_archive
+        - wheel
         - tk  # [linux]
       run:
         - python
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:44]: `contourpy>=1.0.1` is required in `run`, see https://github.com/matplotlib/matplotlib/blob/318b2348f7127efc3e28aadebf762aae32b6343c/setup.py#L309

File: recipe/meta.yaml [hunk 4]
```diff
@@ -38,6 +38,7 @@ outputs:
         - setuptools
         - setuptools_scm >=4
         - setuptools_scm_git_archive
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:40]: `setuptools_scm_git_archive` is no more needed, see https://github.com/matplotlib/matplotlib/commit/a31a01623283f3672b79dc2736957b4fd49ebe07

File: recipe/meta.yaml (hunk 2 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:15]: Conda-forge made changes in `requirements/host` for v3.5.3, see https://github.com/conda-forge/matplotlib-feedstock/commit/5cdeffcebbf5c4a05ef099936a0df900ebfc38c5. Could you verify if it's suitable for us?

Comment (marcoesters) [reply recipe/meta.yaml:15]: Not for the version in question though: https://github.com/matplotlib/matplotlib/blob/v3.5.3/setup.py#L11

When I download the source code, it also says 3.7.

Comment (skupr-anaconda) [reply recipe/meta.yaml:15]: Ah, I see. Thanks!

**Review by marcoesters — COMMENTED** [review]
What is the rationale for all the numpy version logic?

I also don't see freetype required upstream. Where does that requirement come from?

It would be great if these question could be addressed as comments in the recipe.

File: recipe/meta.yaml (hunk 2 above)
Comment (varlackc) [reply recipe/meta.yaml:15]: The minimum requirement for `python 3.8` seems to be for `Matplotlib 3.6` and above.
https://github.com/matplotlib/matplotlib/blob/318b2348f7127efc3e28aadebf762aae32b6343c/setup.py#L12

For `Matplotlib 3.5.3` the minnimum python version seems to be still `python 3.7`
https://github.com/matplotlib/matplotlib/blob/v3.5.3/setup.py#L11

File: recipe/meta.yaml (hunk 3 above)
Comment (Arishamays1) [reply recipe/meta.yaml:44]: Thanks, Serhii

**Comment by Arishamays1** [comment]
> What is the rationale for all the numpy version logic?
> 
> I also don't see freetype required upstream. Where does that requirement come from?
> 
> It would be great if these question could be addressed as comments in the recipe.

Hey @skupr-anaconda, just wanted to know more about numpy versions in the meta.yaml.

**Comment by skupr-anaconda** [comment]
> > What is the rationale for all the numpy version logic?
> > I also don't see freetype required upstream. Where does that requirement come from?
> > It would be great if these question could be addressed as comments in the recipe.
> 
> Hey @skupr-anaconda, just wanted to know more about numpy versions in the meta.yaml.
;

@Arishamays1 @marcoesters 
The rationale for `numpy` is the next:
- Not all our platforms have artifacts for the same versions of numpy (e.x., **1.20** versions). That's why it was decided to use `1.19` and `1.21` branches.
- For `host` it's better to use **the oldest supported** numpy. In the case of `matplotlib`, the minimum version is `>=1.17`, see https://github.com/matplotlib/matplotlib/blob/d04c8de4a89ab756a137ef146e8542e81041909f/setup.py#L318
- NumPy version support says that versions `1.17` and `1.18` are already not supported, see https://numpy.org/neps/nep-0029-deprecation_policy.html#drop-schedule. That's another reason do not use them.
- The current general conda_build_config.yaml are not suitable because of `v0.16` for platfroms `not (osx and arm64)`,  see https://github.com/AnacondaRecipes/aggregate/blob/1ce8bd49889aa0496d7e36faa15cde74ed6bcfc9/conda_build_config.yaml#L156-L159

**Comment by marcoesters** [comment]
> > > > What is the rationale for all the numpy version logic?
> > > > I also don't see freetype required upstream. Where does that requirement come from?
> > > > It would be great if these question could be addressed as comments in the recipe.
> > > 
> > > 
> > > Hey @skupr-anaconda, just wanted to know more about numpy versions in the meta.yaml.
> > > ;
> > 
> > 
> > @Arishamays1 @marcoesters The rationale for `numpy` is the next:
> > 
> > * Not all our platforms have artifacts for the same versions of numpy (e.x., **1.20** versions). That's why it was decided to use `1.19` and `1.21` branches.
> > * For `host` it's better to use **the oldest supported** numpy. In the case of `matplotlib`, the minimum version is `>=1.17`, see https://github.com/matplotlib/matplotlib/blob/d04c8de4a89ab756a137ef146e8542e81041909f/setup.py#L318
> > * NumPy version support says that versions `1.17` and `1.18` are already not supported, see https://numpy.org/neps/nep-0029-deprecation_policy.html#drop-schedule. That's another reason do not use them.
> > * The current general conda_build_config.yaml are not suitable because of `v0.16` for platfroms `not (osx and arm64)`,  see https://github.com/AnacondaRecipes/aggregate/blob/1ce8bd49889aa0496d7e36faa15cde74ed6bcfc9/conda_build_config.yaml#L156-L159
> 
> Thanks for the clarification, Serhii! This makes sense.

Could you document this in the `meta.yaml`? A link to Serhii's explanation should be sufficient.

My question about `freetype` still needs to be addressed, too. I didn't find it in the upstream requirements.

**Comment by Arishamays1** [comment]
> What is the rationale for all the numpy version logic?
> 
> I also don't see freetype required upstream. Where does that requirement come from?
> 
> It would be great if these question could be addressed as comments in the recipe.

I noticed that conda-forge also utilizes `FreeType` in their meta.yaml file; however, I'm not 100% sure the reasoning.
https://github.com/conda-forge/matplotlib-feedstock/blob/d29adb917403fc067934aa90a7c2001006ca5e36/recipe/meta.yaml#L38

**Review by varlackc — COMMENTED** [review]
The linter mentions that `doc_source_url` is missing:

The following could be a potential url:

`doc_source_url`
https://github.com/matplotlib/matplotlib/tree/main/doc

**Comment by marcoesters** [comment]
Looking into this more, `freetype` is not a package for testing, it is a C requirement: https://github.com/matplotlib/matplotlib/blob/v3.5.3/doc/devel/dependencies.rst

Matplotlib can supply its own, but we configure it such that we use a system freetype, which we do in the build script.

**Review by marcoesters — APPROVED** [review]

File: recipe/meta.yaml [hunk 5]
```diff
@@ -32,6 +32,8 @@ outputs:
         - pip
         - certifi >=2020.06.20
         - freetype >=2.3
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:34]: We can add a comment about freetype https://github.com/matplotlib/matplotlib/blob/v3.5.3/doc/devel/dependencies.rst#c-libraries

**Review by skupr-anaconda — APPROVED** [review]

**Comment by skupr-anaconda** [comment]
It's a temporary Prefect issue on `win64`. Waiting for when it will be fixed.

**Review by cbouss — APPROVED** [review]
LGTM

**Pull request merged**